packages/core/src/new-api/internal/deployment-loader/memory-deployment-loader.ts

```diff
@@ -0,0 +1,55 @@
+import fs from "fs";
+
+import { Artifact, ArtifactResolver, BuildInfo } from "../../types/artifact";
+import { DeploymentLoader } from "../../types/deployment-loader";
+import { Journal } from "../../types/journal";
+import { MemoryJournal } from "../journal/memory-journal";
+
+export class MemoryDeploymentLoader implements DeploymentLoader {
+  public journal: Journal;
+
+  private _deployedAddresses: { [key: string]: string };
+
+  constructor(private _artifactResolver: ArtifactResolver) {
+    this.journal = new MemoryJournal();
+    this._deployedAddresses = {};
+  }
+
+  public async loadArtifact(storedArtifactPath: string): Promise<Artifact> {
+    const json = await fs.promises.readFile(storedArtifactPath);
```

Comment:
I don't understand this. How does the in memory one read from the fs? I must be missing something.

Reply:
`memory-deployment-loader` is a misnomer. Maybe `ephemeral-deployment-loader`, this effectively a deferral of the artifact loading to within the execution engine; but that loading is from the hardhat artifacts path.

Reply:
I renamed to `ephemeral-deployment-loader`.